smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/member/MemberShapeDecodeGenerator.kt

```diff
@@ -193,4 +233,33 @@ open class MemberShapeDecodeGenerator(
             else -> ""
         }
     }
+
+    private fun resolveBlobDefault(targetShape: Shape, value: String = ""): String {
+        writer.addImport(FoundationTypes.Data)
+        return if (targetShape.hasTrait<StreamingTrait>()) {
+            " ?? ByteStream.data(Data(\"$value\".utf8))"
+        } else {
+            " ?? Data(\"$value\".utf8)"
+        }
+    }
+
+    private fun resolveDocumentDefault(useZeroValue: Boolean, node: Node): String {
+        return when {
+            node.isObjectNode -> { " ?? Document.object([:])" }
+            node.isArrayNode -> { " ?? Document.array([])" }
+            node.isStringNode -> {
+                val resolvedValue = "".takeIf { useZeroValue } ?: node.expectStringNode().value
+                " ?? Document.string(\"$resolvedValue\")"
+            }
+            node.isBooleanNode -> {
+                val resolvedValue = "false".takeIf { useZeroValue } ?: node.expectBooleanNode().value
+                " ?? Document.boolean($resolvedValue)"
+            }
+            node.isNumberNode -> {
+                val resolvedValue = "0".takeIf { useZeroValue } ?: node.expectNumberNode().value
+                " ?? Document.number($resolvedValue)"
+            }
+            else -> { return "" } // null node type means no default value but explicit
```

Comment:
nit: `return` is unnecessary since the return is on 247 outside of when. scope is also unnecessary for single line statements

`else -> ""`

Reply:
Change adopted